cmd/satellite/main.go

```diff
@@ -153,7 +153,7 @@ func cmdSetup(cmd *cobra.Command, args []string) (err error) {
 func cmdDiag(cmd *cobra.Command, args []string) (err error) {
 	// open the psql db
 	dbpath := diagCfg.BasePath
-	dbm, err := bwagreement.NewDBManager("postgres", dbpath, zap.NewNop())
+	dbm, err := bwagreement.NewDBManager(context.Background(), "postgres", dbpath, zap.NewNop())
```

Comment:
I tried to abstract away the database creation into `Config.Run()`, but `NewDBManager` is also called from here. I opted to pass the context into `NewDBManager` instead.

Reply:
The suggestion I made was not just about context, it's also being able to supply the database implementation from outside to the NewServer, separating the concerns.

Reply:
That wasn't obvious. I'll try again harder.

Reply:
Nah, it's my fault for not explaining further why it would be "more cleaner".